be/src/runtime/sender_queue.cpp

```diff
@@ -730,17 +752,23 @@ Status DataStreamRecvr::PipelineSenderQueue::add_chunks(const PTransmitChunkPara
             }
             iter++;
         }
+
         if (!chunks.empty() && done != nullptr && _recvr->exceeds_limit(total_chunk_bytes)) {
             chunks.back().closure = *done;
             chunks.back().queue_enter_time = MonotonicNanos();
             COUNTER_UPDATE(_recvr->_closure_block_counter, 1);
             *done = nullptr;
         }
+
         for (auto& chunk : chunks) {
             int index = _is_pipeline_level_shuffle ? chunk.driver_sequence : 0;
             size_t chunk_bytes = chunk.chunk_bytes;
             auto* closure = chunk.closure;
-            _chunk_queues[index].enqueue(std::move(chunk));
+            if (keep_order && use_pass_through) {
+                _chunk_queues[index].enqueue(*_producer_token, std::move(chunk));
```

Comment:
I don't get why here we should have `producer_token`? to keep order?

Reply:
The chunk enqueue first must be dequeue first either.

If withouth this producer_token, for instance, chunk1 and chunk2 are enqueued in order, but may be chunk2 is dequeue first.